File: PersonnelManagement/PersonnelManagement/Front/classes/persondecreeblocksub.ts
﻿import Personreward from './personreward';
import Person from './person';
import Personphoto from './personphoto';

export default class Persondecreeblocksub {
    id: number;
    persondecree: number;
    persondecreeblock: number;
    persondecreeblockintro: number;
    persondecreeblocksubtype: number;
    intro: number;
    introtext: string;
    index: number;
    priority: number;
    subvaluenumber1: number;
    subvaluenumber2: number;
    subvaluenumber3: number;
    subvaluenumber4: number;
    subvaluestring1: string;
    subvaluestring2: string;
    subvaluestring3: string;
    subvaluestring4: string;
    parentpersondecreeblocksub: number;
    
    subvaluedate1: Date;
    subvaluedate2: Date;
    
    status: number; // 1 - Добавить.


    samplePersonreward: Personreward; // шаблон для добавления 
    fiosearch: string = ""; // Для поиска
    person: Person; // Для поиска
    personssearch: Person[] = []; // Для поиска
    photosPreview: Personphoto[] = []; // Для поиска
    searchiteration: number = 0; // Для поиска
}

File: PersonnelManagement/PersonnelManagement/Models/Persondecreeblocksub.cs
using System;
using System.Collections.Generic;

namespace PersonnelManagement.Models
{
    public partial class Persondecreeblocksub
    {
        public int Id { get; set; }
        public int Persondecree { get; set; }
        public int Persondecreeblock { get; set; }
        public int Persondecreeblockintro { get; set; }
        public int Persondecreeblocksubtype { get; set; }
        public int Intro { get; set; }
        public string Introtext { get; set; }
        public int Index { get; set; }
        public int Priority { get; set; }
        public int Subvaluenumber1 { get; set; }
        public int Subvaluenumber2 { get; set; }
        public string Subvaluestring1 { get; set; }
        public string Subvaluestring2 { get; set; }
        public int Parentpersondecreeblocksub { get; set; }
        public DateTime? Subvaluedate1 { get; set; }
        public DateTime? Subvaluedate2 { get; set; }
    }
}
